src/node_zlib.cc

```diff
@@ -236,6 +239,40 @@ class ZCtx : public AsyncWrap {
         ctx->err_ = deflate(&ctx->strm_, ctx->flush_);
         break;
       case UNZIP:
+        next_expected_header_byte = ctx->strm_.next_in;
+
+        switch (ctx->gzip_id_bytes_read_) {
```

Comment:
Sorry, you’re right – updated again. And no, no bytes are lost in that case; there is always the fall-through in the outer switch from `case UNZIP:` to `case INFLATE:`, so all input ends up being passed to `inflate()`.